src/mongocxx/test/spec/unified_tests/assert.cpp

```diff
@@ -31,6 +32,48 @@ using namespace mongocxx;
 using assert::to_string;
 using bsoncxx::types::bson_value::value;
 
+namespace {
+
+/// When we are matching two documents, we walk down them recursively. To aide test debugging, store
+/// a path as we go, in this variable:
+thread_local std::vector<std::string> S_match_doc_path;
```

Comment:
Nice, this is a cool idea. Is the rationale for the `thread_local` to make a future parallel test runner easier?

Reply:
Yes-ish. While parallel runners in the future would probably use separate OS processes, I like to use `thread_local` to store call-stack-local state Just In Case™ there are ever multiple threads. Passing the path down the call stack would be "the proper way", but I find usage of an internal-linkage thread-local to be cleaner for cases like this.

Reply:
Ack. That seems like a good precaution.